st2api/st2api/controllers/exp/inquiries.py

```diff
@@ -0,0 +1,295 @@
+from st2common.rbac import utils as rbac_utils
+from st2common.router import abort
+from st2common.models.api.execution import ActionExecutionAPI
+from st2common.persistence.execution import ActionExecution
+from st2common.services import action as action_service
+from st2common.util.action_db import (get_action_by_ref, get_runnertype_by_name)
+
+
+from st2actions.container.base import get_runner_container
+
+__all__ = [
+    'InquiriesController'
+]
+
+LOG = logging.getLogger(__name__)
+
+
+class InquiriesController(ResourceController):
+    """API controller for Inquiries
+    """
+
+    supported_filters = copy.deepcopy(SUPPORTED_FILTERS)
+
+    # No data model currently exists for Inquiries, so we're "borrowing" ActionExecutions
+    model = ActionExecutionAPI
+    access = ActionExecution
+
+    def get_all(self, requester_user=None, limit=None, **raw_filters):
+        """Retrieve multiple Inquiries
+
+           TODO(mierdin): This function doesn't currently make RBAC assertions for two reasons:
+           - The equivalent for executions (Resource._get_all) doesn't seem to do this either.
+           - I am not sure the best approach, since I would have to make an assertion
+             inside the below loop?
+
+            Handles requests:
+                GET /inquiries/
+        """
+
+        raw_inquiries = super(InquiriesController, self)._get_all(
+            limit=limit,
+            raw_filters={'status': 'pending', 'runner': 'inquirer'}
+        )
+
+        inquiries = []
+        for raw_inquiry in json.loads(raw_inquiries.body):
+
+            # _get_all includes workflows that also contain executions
+            # with 'pending' status, so we want to prune all workflows
+            # (The action won't have a 'children' key)
+            if raw_inquiry.get('children'):
+                continue
+
+            inquiries.append(self._transform_inquiry(raw_inquiry))
+
+        return inquiries
+
+    def get_one(self, inquiry_id, requester_user=None):
+        """Retrieve a single Inquiry
+
+            Handles requests:
+                GET /inquiries/<inquiry id>
+        """
+
+        # Retrieve the desired inquiry
+        #
+        # (Passing permission_type here leverages _get_one_by_id's built-in
+        # RBAC assertions)
+        raw_inquiry = self._get_one_by_id(
+            id=inquiry_id,
+            requester_user=requester_user,
+            permission_type=PermissionType.INQUIRY_VIEW
+        )
+
+        if raw_inquiry.runner.get('runner_module') != "inquirer":
+            abort(http_client.BAD_REQUEST, '%s is not an Inquiry.' % inquiry_id)
+            return
+
+        if raw_inquiry.status != "pending":
+            abort(http_client.BAD_REQUEST, 'Inquiry %s has already been responded to' % inquiry_id)
+            return
+
+        new_inquiry = self._transform_inquiry(raw_inquiry)
+
+        return new_inquiry
+
+    def put(self, inquiry_id, response_data, requester_user):
```

Comment:
Same here, we should also utilize API model for input / request data.

Reply:
+1 to kami's comment

Reply:
Hrm, so I would have to create a `st2common.models.api.inquiry` for this - wouldn't I have to also create a DB model for Inquiries, since the API models seem to refer to the DB models?

Reply:
And what about `st2common.persistence.inquiry`?

Reply:
Nvm, I think I can see how I can just create a new `st2common.models.api.inquiry`, but it's already taken some tinkering and will require some more. Will take a shot at this and let you know what I come up with.

Reply:
@Kami @lakshmi-kannan Okay I pushed https://github.com/StackStorm/st2/pull/3653/commits/920e88f81777eff5d0e5dbf6f86fe21f1f814415 as a first attempt at this. It seems to work. It's a **bit** unorthodox since I'm only implementing the API models, but I think it's a good first step. Let me know if you see anything heinous.